### PR TITLE
Handle power9 compat mode workaround with pvr check

### PR DESCRIPTION
**Pull request by sathnaga**
Power9 compat mode workaround is not needed for
processor revisions >= DD2.2, hence this patch
handles that condition and removes unwanted params
for smt switch and move whole logic check for smt
changes for only powerpc cpus.

Signed-off-by: Satheesh Rajendran <sathnaga@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by sathnaga** [comment]
```
avocado run virsh.domtime.positive.managedsave_vm --vt-type libvirt --vt-extra-params create_vm_libvirt=yes kill_vm_libvirt=yes env_cleanup=yes smp take_regular_screendumps=no backup_image_before_testing=no libvirt_controller=virtio-scsi scsi_hba=virtio-scsi-pci drive_format=scsi-hd use_os_variant=no restore_image_afteresting=no vga=none display=nographic power9_compat="yes" restore_smt="yes" --vt-guest-os Pegas.1.1.ppc64le
JOB ID     : 3891ed64697a2e838ba81c550494cfa05e6370f6
JOB LOG    : /home/sath/avocado-fvt-wrapper/results/job-2018-06-18T15.37-3891ed6/job.log
 (1/1) type_specific.powerkvm-libvirt.virsh.domtime.positive.managedsave_vm: PASS (96.95 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 111.31 s


# lscpu
Architecture:          ppc64le
Byte Order:            Little Endian
CPU(s):                160
On-line CPU(s) list:   0-159
Thread(s) per core:    4
Core(s) per socket:    20
Socket(s):             2
NUMA node(s):          2
Model:                 2.2 (pvr 004e 1202)
Model name:            POWER9, altivec supported
CPU max MHz:           3800.0000
CPU min MHz:           2166.0000
L1d cache:             32K
L1i cache:             32K
L2 cache:              512K
L3 cache:              10240K
NUMA node0 CPU(s):     0-79
NUMA node8 CPU(s):     80-159
```

without patch:
```
2018-06-18 15:31:00,709 utils_env        L0088 WARNI| Creating new, empty env file
2018-06-18 15:31:00,717 process          L0554 INFO | Running 'cat /sys/module/kvm_hv/parameters/indep_threads_mode'
2018-06-18 15:31:00,720 process          L0386 DEBUG| [stdout] Y
2018-06-18 15:31:00,721 process          L0642 INFO | Command 'cat /sys/module/kvm_hv/parameters/indep_threads_mode' finished with 0 after 0.000977993011475s
2018-06-18 15:31:00,749 process          L0554 INFO | Running 'echo N > /sys/module/kvm_hv/parameters/indep_threads_mode'
2018-06-18 15:31:00,751 process          L0642 INFO | Command 'echo N > /sys/module/kvm_hv/parameters/indep_threads_mode' finished with 0 after 0.000622987747192s
2018-06-18 15:31:00,752 test_setup       L2270 DEBUG| indep_thread_mode turned N successfully
2018-06-18 15:31:00,752 process          L0554 INFO | Running 'ppc64_cpu --smt'
2018-06-18 15:31:00,824 process          L0386 DEBUG| [stdout] SMT=4
2018-06-18 15:31:00,824 process          L0642 INFO | Command 'ppc64_cpu --smt' finished with 0 after 0.0698189735413s
2018-06-18 15:31:00,848 process          L0554 INFO | Running 'ppc64_cpu --smt=off'
2018-06-18 15:31:17,937 process          L0642 INFO | Command 'ppc64_cpu --smt=off' finished with 0 after 17.0870039463s
2018-06-18 15:31:17,938 test_setup       L2326 DEBUG| SMT turned off successfully
2018-06-18 15:31:17,938 process          L0554 INFO | Running 'true'
2018-06-18 15:31:17,943 process          L0642 INFO | Command 'true' finished with 0 after 0.00110006332397s
2018-06-18 15:31:17,943 process          L0554 INFO | Running 'ps -o comm 1'
2018-06-18 15:31:18,010 process          L0386 DEBUG| [stdout] COMMAND


```
with patch:
```
2018-06-18 15:37:58,115 utils_env        L0088 WARNI| Creating new, empty env file
2018-06-18 15:37:58,123 process          L0554 INFO | Running 'grep revision /proc/cpuinfo | awk '{print $3}' | head -n 1'
2018-06-18 15:37:58,142 process          L0386 DEBUG| [stdout] 2.2
2018-06-18 15:37:58,143 process          L0642 INFO | Command 'grep revision /proc/cpuinfo | awk '{print $3}' | head -n 1' finished with 0 after 0.017560005188s
2018-06-18 15:37:58,154 process          L0554 INFO | Running 'true'
2018-06-18 15:37:58,158 process          L0642 INFO | Command 'true' finished with 0 after 0.000591039657593s
2018-06-18 15:37:58,158 process          L0554 INFO | Running 'ps -o comm 1'
2018-06-18 15:37:58,222 process          L0386 DEBUG| [stdout] COMMAND
2018-06-18 15:37:58,222 process          L0642 INFO | Command 'ps -o comm 1' finished with 0 after 0.0610401630402s
2018-06-18 15:37:58,223 process          L0386 DEBUG| [stdout] systemd
```

File: virttest/env_process.py [hunk 1]
```diff
@@ -648,36 +648,44 @@ def preprocess(test, params, env):
...
+            if "power8" in remote_cpu:
                 test_setup.switch_smt(state="off", params=params)
-            elif "power9" in cmd_output and power9_compat_remote:
+            elif "power9" in remote_cpu and power9_compat_remote and remote_pvr < 2.2:
                 test_setup.switch_indep_threads_mode(state="N", params=params)
                 test_setup.switch_smt(state="off", params=params)
-
+            if pvr != remote_pvr:
+                logging.warning("Source and destinations system PVR "
+                                "does not match")
```
Comment (balamuruhans) [inline virttest/env_process.py]: if we can print the pvr versions of source and destination systems in log it will be helpful.

Comment (sathnaga) [reply virttest/env_process.py]: Sure, it can be done..

Comment (sathnaga) [reply virttest/env_process.py]: @balamuruhans done

**Comment by balamuruhans** [comment]
@sathnaga Thanks, let's get it in.

**Pull request merged**